src/shared/utils.ts

```diff
@@ -0,0 +1,13 @@
+export const waitForElement = async (selector: any, timeout = 5000) => {
+	const startTime = Date.now();
+
+	while (document.querySelector(selector) === null) {
+		if (Date.now() - startTime > timeout) {
+			throw new Error(`Timeout waiting for element with selector: ${selector}`);
+		}
+
+		await new Promise((resolve) => requestAnimationFrame(resolve));
+	}
```

Comment:
I don't think `while` awaits promises

Reply:
I went with the recursive approach and it feels like it loads more quickly.